docs/contrib/06_new_project.md

```diff
@@ -0,0 +1,40 @@
+# Adding a project to pySDC (and automatic testing)
+
+When adding a new project to pySDC, it needs to be tested.
+In order to run those tests as fast as possible, a few steps are needed, that are given below.
+
+## Add a directory in 'pySDC/projects'
+
+First, create a new directory in `pySDC/projects` with the name of your project.
+The code of the new project should go into that newly created directory.
+
+## Add an environment-file
+
+In addition to the codde of your projects, an environment-file is needed.
+This includes the dependencies that are required for your project to run.
+The file needs to be named `env_update.yml` and needs to follow the structure shown below:
+
+```yaml
+name: pySDC
+channels:
+  - conda-forge
+  - defaults
+dependencies:
+  - numpy
+```
+
+The list of dependencies can be extended as needed.
+The name should stay `pySDC`. The channels cover most of the usual packages.
+If a package is needed that cannot be found in those channels by conda (or mamba),
+please add the correct channel to the list.
+
+## Add tests to the project
+
+In order to automatically find the tests of your project, please add the tests to a subdirectory called `tests` in the directory of your project. Furthermore, the files should start with `test_` and the functions should also start with that. For more information see the documentation of pytest on [test discovery](<https://docs.pytest.org/en/8.2.x/explanation/goodpractices.html#tests-as-part-of-application-code>).
+
+## Add the project in the CI-file
+
+To run the tests of all projects in parallel, the projects are explicitly mentioned in the CI-file.
+In order to run the tests of your project, please add the name of your project **as your directory is named**
+in the [CI-File](./../../.github/workflows/ci_pipeline.yml)
+in the job `project_cpu_tests_linux` in the list `strategy/matrix/env`.
```

Comment:
@pancetta, do you want to add a few lines on how to obtain a release of pySDC to go along with a publication?

Reply:
Something like:
```
If you project is published and you need a dedicated pySDC version with a DOI, please get in touch with us and/or open a new issue. We will help you with this as soon as possible. Note that a final DOI is usually only necessary once a paper is accepted and the final proofs are due. We strongly encourage to describe and cite the current version of pySDC already during initial submission, though.
```

Reply:
Added the text proposed by @pancetta in commit 1a4b58f591c1e0a0f1fe1e22922c3f16a81f046c and also added a headline as it is not directly related to the CI-Pipeline